sensor/common/clusterentities/store_container_id.go

```diff
@@ -0,0 +1,175 @@
+package clusterentities
+
+import (
+	"fmt"
+
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/sensor/common/clusterentities/metrics"
+	"golang.org/x/exp/maps"
+)
+
+type containerIDsStore struct {
+	mutex sync.RWMutex
+	// entitiesMemorySize defines how many ticks old endpoint data should be remembered after removal request
+	// Set to 0 to disable memory
+	entitiesMemorySize uint16
+
+	// containerIDMap maps container IDs to container metadata
+	containerIDMap map[string]ContainerMetadata
+	// reverseContainerIDMap maps deployment ids to sets of container IDs associated with this deployment.
+	reverseContainerIDMap map[string]map[string]struct{}
```

Comment:
That is a good suggestion! I am actually reimplementing some parts to use Sets instead of maps and that allowed me to find two tiny bugs in my implementation!